cmd/provider/main.go

```diff
@@ -55,24 +61,44 @@ func main() {
 		ctrl.SetLogger(zl)
 	}
 
-	log.Debug("Starting", "sync-period", syncPeriod.String())
+	log.Debug("Starting", "sync-period", syncInterval.String())
 
 	cfg, err := ctrl.GetConfig()
 	kingpin.FatalIfError(err, "Cannot get API server rest config")
 
-	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
-		LeaderElection:   *leaderElection,
-		LeaderElectionID: "crossplane-leader-election-provider-sql",
+	mgr, err := ctrl.NewManager(ratelimiter.LimitRESTConfig(cfg, *maxReconcileRate), ctrl.Options{
+		// SyncPeriod in ctrl.Options has been removed since controller-runtime v0.16.0
+		// The recommended way is to move it to cache.Options instead
 		Cache: cache.Options{
-			SyncPeriod: syncPeriod,
+			SyncPeriod: syncInterval,
 		},
+
+		// controller-runtime uses both ConfigMaps and Leases for leader
```

Comment:
Could you split out this as a separate PR?

(After we get the CI fixes in)

Reply:
removed!